### PR TITLE
test: add tests for finch cp

### PR DESCRIPTION
**Pull request by mharwani**
Signed-off-by: Mrudul Harwani <mharwani@amazon.com>

Issue #, if available:
runfinch/finch#82

*Description of changes:*
Adding e2e tests for finch cp command

*Testing done:*
Tests passed locally

- [X] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/cp.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	actualContent, err := os.ReadFile(filepath.Clean(path))
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	gomega.Expect(string(actualContent)).To(gomega.Equal(content))
+}
+
+func fileShouldNotExist(path string) {
+	gomega.Expect(path).ToNot(gomega.BeAnExistingFile())
+}
+
+func fileShouldExistInContainer(o *option.Option, containerName string, path string, content string) {
```
Comment (azhouwd) [inline tests/cp.go:140]: nit:
```suggestion
func fileShouldExistInContainer(o *option.Option, containerName, path, content string) {
```
Same as other functions.

File: tests/cp.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", containerResource, path)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("expected container status running"))
+			})
+		})
+	})
+}
+
+func fileShouldExist(path string, content string) {
```
Comment (azhouwd) [inline tests/cp.go:129]: nit: maybe we can move these functions to `tests.go`? These seem to be generic and folks and refer to all the generic ones in `tests.go`.

File: tests/cp.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	gomega.Expect(path).To(gomega.BeARegularFile())
```
Comment (azhouwd) [inline tests/cp.go:130]: Could we use `gomega.BeAnExistingFile` here as well?

File: tests/cp.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+				fileDir := ffs.CreateTempDir("finch-test")
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				command.Run(o, "cp", containerResource, path)
+				fileShouldExist(path, content)
+			})
+
+			for _, link := range []string{"-L", "--follow-link"} {
+				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from host to container and follow symbolic link",
```
Comment (azhouwd) [inline tests/cp.go:59]: The variable needs to be redefined here: more info - https://github.com/runfinch/tests/pull/15#discussion_r984021849
```suggestion
			for _, link := range []string{"-L", "--follow-link"} {
			         link := link
				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from host to container and follow symbolic link",
```

File: tests/cp.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+			})
+		})
+
+		ginkgo.Context("when the container is not running", func() {
+			ginkgo.It("should not be able to copy file from host to container", func() {
+				command.Run(o, "run", "--name", testContainerName, defaultImage)
+				path := ffs.CreateTempFile(filename, content)
+				ginkgo.DeferCleanup(os.RemoveAll, filepath.Dir(path))
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", path, containerResource)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("expected container status running"))
```
Comment (azhouwd) [inline tests/cp.go:113]: Currently if we want to verify the text output, we only verify something we can control. These texts are from upstream and we can't easily control what does it look like. I think `command.RunWithoutSuccessfulExit(o, "cp", path, containerResource)` should be enough. Suggest removing output validation from upstream for now.

Comment (mharwani) [reply tests/cp.go:113]: Makes sense. I will also remove the checks for `finch cp` failure cases.

File: tests/cp.go (hunk 3 above)
Comment (mharwani) [reply tests/cp.go:130]: I think `BeARegularFile` is more accurate as we are not checking for other file types like directories.

**Pull request merged**